libcgroup: update to 2.0.

wibed opened the pull request:
update libcgroup to version 2.0

<!-- Mark items with [x] where applicable -->

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->
<!-- 
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64-musl)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl
-->

paper42 commented on srcpkgs/libcgroup/patches/musl-decls.patch:
```diff
@@ -18,5 +18,5 @@ include file sys/cdefs.h
 +
  #define _LIBCGROUP_H_INSIDE
- 
+
```
why?

wibed replied:
cp paste creep

paper42 commented on srcpkgs/libcgroup/patches/musl-sterror_r.patch:
```diff
@@ -4,7 +4,7 @@ GNU libc's if _GNU_SOURCE is defined.
 +++ b/src/api.c	2015-10-25 14:09:18.690232596 +0100
 @@ -3332,8 +3332,16 @@
- 
+
```
why?

wibed replied:
cp paste creep

paper42 commented on srcpkgs/libcgroup/template:
```diff
@@ -1,18 +1,16 @@
-distfiles="${SOURCEFORGE_SITE}/libcg/${pkgname}-${version}.tar.bz2"
-checksum=e4e38bdc7ef70645ce33740ddcca051248d56b53283c0dc6d404e17706f6fb51
-
+homepage="https://github.com/${pkgname}/${pkgname}"
```
don't use $pkgname in homepage

wibed replied:
pgkname removed from var

paper42 commented on srcpkgs/libcgroup/template:
```diff
@@ -1,18 +1,16 @@
-checksum=e4e38bdc7ef70645ce33740ddcca051248d56b53283c0dc6d404e17706f6fb51
-
+homepage="https://github.com/${pkgname}/${pkgname}"
+distfiles="${homepage}/releases/download/v2.0/${pkgname}-${version}.tar.bz2"
```
I wouldn't use $homepage and $pkgname. 2.0 should be replaced with $version. tar.gz is prefered over bz2.

wibed replied:
i kept $homepage and $pkgname because it shortens the url quite a bit

paper42 replied:
I don't think we have a shortage of bytes.

wibed replied:
it would be nice to keep the 80 column mark as close as possible.

paper42 replied:
distfiles often go over that mark and that's fine.

wibed replied:
distfile now contains full url

paper42 commented:
pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.

paper42 commented on srcpkgs/libcgroup/patches/musl-sterror_r.patch:
```diff
@@ -17,6 +16,6 @@ GNU libc's if _GNU_SOURCE is defined.
 +	}
 +#endif
- 
+
```
please revert the changes on patches

wibed replied:
the file has been reverted as in the original

paper42 replied:
Something is wrong, GitHub still shows the change. Are you sure you reverted the changes?

paper42 commented on srcpkgs/libcgroup/template:
```diff
@@ -1,18 +1,16 @@
-
+homepage="https://github.com/libcgroup/libcgroup"
+distfiles="${homepage}/releases/download/v${version}/${pkgname}-${version}.tar.gz"
+checksum=aecc501a9ea6a97da0673585db5081df912ae607dec36d5f6f7ab14f69d48ab8
 case "$XBPS_TARGET_MACHINE" in
```
add an empty line between checksum= and case.

wibed replied:
added

paper42 replied:
this was not solved

paper42 replied:
this was not solved

sorry for the double comment

wibed commented:
> pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.

do you mean the block? 
```
pre_configure() {
  autoreconf -if
}
```

paper42 commented:
> > pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.
> 
> do you mean the block?
> 
> ```
> pre_configure() {
>   autoreconf -if
> }
> ```

yes and you might be able to remove the hostmakedepends variable then

wibed commented:
> > > pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.
> > 
> > 
> > do you mean the block?
> > ```
> > pre_configure() {
> >   autoreconf -if
> > }
> > ```
> 
> yes and you might be able to remove the hostmakedepends variable then

without the pre_configure block it builds fine
but it depends on the hostmakedepends variable

paper42 commented:
> > > pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.
> > 
> > yes and you might be able to remove the hostmakedepends variable then
> 
> without the pre_configure block it builds fine
> but it depends on the hostmakedepends variable

sorry, I forgot about flex, we can drop automake and libtool, but have to leave flex there

```
hostmakedepends="flex"
```

Can you also fix the lint issue from CI?

> srcpkgs/libcgroup/template:10: use SPDX id for 'LGPL-2.1' license or see Manual.md

paper42 commented:
@wibed are you still interested in continuing to work on this PR?

wibed commented:
> @wibed are you still interested in continuing to work on this PR?

@paper42 i do, please be patient.

wibed commented:
@mt3 you just comitted literally when i was to to push my changes upstream

wibed commented:
@mt3 it seems the checksum should be updated

paper42 reviewed (commented):
please squash the commits to one and rebase on top of master, so CI can run

paper42 commented:
> @mt3 it seems the checksum should be updated

yes, that should fix the CI

paper42 commented on srcpkgs/libcgroup/template:
```diff
@@ -1,17 +1,16 @@
-checksum=e4e38bdc7ef70645ce33740ddcca051248d56b53283c0dc6d404e17706f6fb51
+license="LGPL-2.1-only"
+homepage="https://github.com/libcgroup/libcgroup"
+distfiles="https://github.com/libcgroup/libcgroup/releases/download/v${version}/libcgroup-${version}.tar.bz2"
```
let's use tar.gz

paper42 reviewed (commented):
squash the commits and it should be good to go

wibed commented:
okey, despise the "no push" commit message. i have reverted the commit back to 2a257f4, because i have committed and pushed from root directory. after that i have rebased to master and pushed up again.

wibed commented:
it is getting late

ericonr commented:
How are you trying to rebase and squash the commits? We have a lot of comments in PRs here explaining how to do it correctly, if you need the exact commands.

wibed commented:
@ericonr 

i usually do a : `git rebase -i HEAD~N` .

but i get all the other commits, because i am not in my own branch.
overreaching my pr.

is there another way i am not aware of?

wibed commented:
ill reopen a new pr
